# Exploratory findings and limitations

Real PromptCloud Flipkart listings. Core transformations are shared with src/ and SQL files. Run the pipeline first.

In [1]:
from pathlib import Path
import os
ROOT = Path.cwd()
if ROOT.name == 'notebooks':
    ROOT = ROOT.parent
os.chdir(ROOT)
import duckdb
import pandas as pd


In [2]:
categories = pd.read_csv('reports/tables/categories.csv')
print(categories[['category','listings','p25','p50','p75','mean_discount_pct']].head(10).to_string(index=False))

                  category  listings   p25    p50     p75  mean_discount_pct
                  Clothing      6198 374.0  499.0  799.00          45.070525
                 Jewellery      3531 310.0  449.0 1076.75          40.889934
                  Footwear      1227 499.0  899.0 1540.00          36.268531
     Mobiles & Accessories      1099 449.0  599.0 1499.00          50.994470
                Automotive      1012 499.0 1350.0 1599.00          54.650991
Home Decor & Festive Needs       929 426.5  699.0 1250.00          36.649404
  Beauty and Personal Care       710 299.0  355.0  479.00          19.829674
           Home Furnishing       700 523.0  799.0 1332.00          37.218050
          Kitchen & Dining       647 199.0  299.0  582.00          43.771695
                 Computers       578 350.0 1250.0 1995.00          43.467316


In [3]:
print(Path('reports/findings.md').read_text())

# Findings

- Clothing accounts for 31.0% of retained listings.
- Sunglasses has the highest mean discount among categories with at least 30 valid discounts: 60.6%.
- 1,849 listings (9.2%) have a valid numeric rating. Rating results describe this subset only.
- 1,919 listings exceed the category IQR fences. These are candidates for manual review, not confirmed pricing errors.
- Brand is missing for 5,861 listings (29.3%). Brand rankings are incomplete.

Prices are INR. Historical crawl: 2015-12-01 06:13:00+00:00 to 2016-06-28 08:45:50+00:00.

## Recommended next steps

Review flagged items within narrower product types and check variant, unit, and bundle differences. Compare discount policies only after checking category composition. Collect sales, costs, stock, and time-series data before assessing profitability or proposing price changes.

No sales volumes, costs, realized transaction prices, or representative sampling weights are available.



## Category prices
![Price percentiles](../reports/figures/price_ranges.png)

## Discounts
![Category discounts](../reports/figures/discounts.png)

These historical listing-level patterns do not measure sales, profit, or causal price effects. Ratings cover only 9.2% of records.